docs/content/api-overview/resources/functions.md

```diff
@@ -45,6 +45,8 @@ The Functions builder is used to create Azure Functions accounts. It abstracts t
 | publish_as | Specifies whether to publish function as code or as a docker container. |
 | add_slot | Adds a deployment slot to the app |
 | add_slots | Adds multiple deployment slots to the app |
+| slot_setting | Sets a deployment slot setting of the function in the form "key" "value". |
```

Comment:
I wonder whether it's possible to piggy back on the existing `setting` keywords? Also, there's also the `secret_setting` keyword for adding secure settings - this should probably be migrated as well if we keep with separate keywords.

Reply:
To make settings sticky, their names should be specified in webapp(`sites`) child resource, `slotconfignames`. We usually add settings in webapp, so the idea was to have `slot_setting `which adds the setting and creates child resource `slotconfignames`. 

The alternative solution was something similar to this:

```
let myWebApp = webApp {
    name "test"
    add_slot deploymentSlot

    settings ["my_setting", "test value"
                  "my_slot_setting", "sticky value"]

    slot_setting_names ["my_slot_setting" ]
}
```
Where `slot_setting_names` was only marking the settings as sticky by adding `slotconfignames`, But we thought the current solution with `slot_setting` will make things clearer.